public_src/services/route-master-wizard.service.ts

```diff
@@ -0,0 +1,270 @@
+    }
+    for (let res of this.savedMultipleNodeDataResponses) {
+      this.processSrv.processNodeResponse(res);
+    }
+    for (let res of this.savedMasterQueryResponses) {
+      this.processSrv.processMastersResponse(res);
+    }
+  }
+
+  /**
+   * Renders data on modal map
+   * @param transformedGeoJSON
+   * @param {Map} map
+   */
+  public renderTransformedGeojsonDataRMWizard(transformedGeoJSON: any, map: L.Map): void {
+    this.ptLayerModal = L.geoJSON(transformedGeoJSON, {
+      filter       : (feature) => {
+        if (!this.elementsRenderedModalMap.has(feature.id) &&
+          'public_transport' in feature.properties && feature.id[0] === 'n'
+        ) {
+          return true;
+        } else {
+          return false;
+        }
+      },
+      onEachFeature: (feature, layer) => {
+        this.elementsRenderedModalMap.add(feature.id);
+        this.enableClickForRouteMasterWizardMap(feature, layer);
+      },
+      pointToLayer : (feature, latlng) => {
+        return this.mapSrv.stylePoint(feature, latlng);
+      },
+    });
+    console.log('LOG (route master wizard s.) Adding PT layer to modal map again', this.ptLayerModal);
+    this.ptLayerModal.addTo(map);
+  }
+
+  /***
+   * Enables click of nodes for modal map
+   * @param feature
+   * @param layer
+   * @returns {void}
+   */
+  public enableClickForRouteMasterWizardMap(feature: any, layer: any): void {
+    layer.on('click', () => {
+      this.handleRouteMasterWizardMarkerClick(feature);
+    });
+  }
+
+  /***
+   * Handles map click
+   * @param feature
+   * @returns {void}
+   */
+  private handleRouteMasterWizardMarkerClick(feature: any): void {
+    const featureId: number = this.mapSrv.getFeatureIdFromMarker(feature);
+    this.autoRouteMapNodeClick.emit(featureId);
+  }
+
+  public findToBeComparedRels(response: any): any {
+    let newDownloadedRoutes = [];
+    let oldDownloadedRoutes = [];
+    if (response) {
+      for (let element of response['elements']) {
+        if ((element.type === 'relation')
+          && !(element.tags.public_transport === 'stop_area'
+            && element.tags.public_transport === 'ref')
+          && (element.members)) {
+          newDownloadedRoutes.push(element);
+        }
+      }
+    }
+    this.modalMapElementsMap.forEach((element) => {
+      if ((element.type === 'relation')
+        && !(element.tags.public_transport === 'stop_area')
+        && element.tags.ref
+        && (element.members)
+        && this.checkMembersInBounds(element)) {
+        oldDownloadedRoutes.push(element);
+      }
+    });
+    console.log('LOG (route master wizard s.) Newly downloaded relations',
+      newDownloadedRoutes, 'old already present relations', oldDownloadedRoutes);
+
+    let rels      = newDownloadedRoutes.concat(oldDownloadedRoutes);
+    let relsMap   = new Map();
+    let refOfRels = [];
+    rels.forEach((rel) => {
+      let noOfMembers     = rel['members'].length;
```

Comment:
do not use [] for parameters, it is not checked in TS, add interface `IPtRelation` and use `.members`